### ISSUE TITLE
checking unbounded quantification

### ISSUE THREAD

**Issue by Laplace-Demon** [OCamlPro/owi#425]
[discussion 23/08](https://github.com/Laplace-Demon/E-ACSL-examples/blob/main/checking_unbounded_quantification.md)

I don't know if it's feasible for the solver

**Comment by zapashcanon** [comment]
Do you want to start implementing it ?

In the meantime, we can probably take care of providing the missing `model_exists(var, prop)` function, although I'm not completely sure how we could implement it because it is more complicated than simply asking if a model exists. (cc @filipeom)

**Comment by Laplace-Demon** [comment]
> Do you want to start implementing it ?
> 
> In the meantime, we can probably take care of providing the missing `model_exists(var, prop)` function, although I'm not completely sure how we could implement it because it is more complicated than simply asking if a model exists. (cc @filipeom)

There won't be much problem with respect to code generation, and I've just realized we no longer need the PNF form, a simple recursion works.

> - `model_exists(var, prop)` which returns a boolean, indicating whether the proposition prop holds true for one possible value of var.
> - `model_forall(var, prop)` which returns a boolean, indicating whether the proposition prop holds true for all possible values of var.

Meanwhile, I'm not sure if those bring difficulty to the solver?

**Comment by zapashcanon** [comment]
No, we actually want to use Z3! We are going to craft an example of what we want to achieve and show it to you on Friday :)
